src/utils/hydration.js

```diff
@@ -142,19 +142,85 @@ function injectRepeatedRows({ step, repeatedRowCount }) {
   };
 }
 
+const formatTrackValue = (value, type = 'col', isEmpty) => {
+  if (isEmpty && value === 'auto') value = '0'; // Default empty value
+  if (type === 'col') return `minmax(0, ${value})`;
+  return value;
+};
+
+const formatDimensionValue = (value, type = 'col') => {
+  switch (value) {
+    case 'fill':
+      return type === 'col' ? '100%' : 'fit-content';
+    case 'auto':
+      return type === 'col' ? 'auto' : 'fit-content';
+    default:
+      return parseInt(value);
+  }
+};
+
+const getTrackElementsMap = (step) => {
+  const {
+    servar_fields: fields,
+    progress_bars: progressBars,
+    images,
+    texts,
+    buttons
+  } = step;
+
+  const allElements = [
```

Comment:
i think u should decompose this logic with this logic here https://github.com/feathery-org/feathery-react/blob/master/src/form/Form.jsx#L188